project-chip/connectedhomeip
# Disable faultinjection cluster in Darwin and Android API surface

#### Problem
What is being fixed?  Examples:
* https://github.com/project-chip/connectedhomeip/pull/21951 added the FaultInjection cluster to the API surface for Darwin and Android.  faultinjection is mainly used to exercises various negative path to improve current test coverage, and gcoverage is only run against linux platform now,  so we should disable it on Darwin and Android.

#### Change overview
Disable faultinjection cluster in Darwin and Android API surface

#### Testing
How was this tested? (at least one bullet point required)
* Build all-clusters-app with nlfaultinjection enabled
`scripts/examples/gn_build_example.sh examples/all-clusters-app/linux out/debug/standalone chip_config_network_layer_ble=false chip_with_nlfaultinjection=true`

* Configure a system fault from chip_tool
`./chip-tool faultinjection fail-at-fault 1 1 0 1 true 12344321 0`

* Confirm the fault is activated from all-clusters-app log

```
[1661909717.710111][338693:338693] CHIP:EM: Received message of type 0x8 with protocolId (0, 1) and MessageCounter:33050415 on exchange 15984r
[1661909717.710191][338693:338693] CHIP:EM: Handling via exchange: 15984r, Delegate: 0x556a3a6c4b48
[1661909717.710310][338693:338693] CHIP:DMG: InvokeRequestMessage =
[1661909717.710343][338693:338693] CHIP:DMG: {
[1661909717.710367][338693:338693] CHIP:DMG:    suppressResponse = false, 
[1661909717.710393][338693:338693] CHIP:DMG:    timedRequest = false, 
[1661909717.710422][338693:338693] CHIP:DMG:    InvokeRequests =
[1661909717.710466][338693:338693] CHIP:DMG:    [
[1661909717.710492][338693:338693] CHIP:DMG:        CommandDataIB =
[1661909717.710519][338693:338693] CHIP:DMG:        {
[1661909717.710543][338693:338693] CHIP:DMG:            CommandPathIB =
[1661909717.710579][338693:338693] CHIP:DMG:            {
[1661909717.710612][338693:338693] CHIP:DMG:                EndpointId = 0x0,
[1661909717.710646][338693:338693] CHIP:DMG:                ClusterId = 0xfff1fc06,
[1661909717.710678][338693:338693] CHIP:DMG:                CommandId = 0x0,
[1661909717.710709][338693:338693] CHIP:DMG:            },
[1661909717.710744][338693:338693] CHIP:DMG:            
[1661909717.710774][338693:338693] CHIP:DMG:            CommandFields = 
[1661909717.710806][338693:338693] CHIP:DMG:            {
[1661909717.710840][338693:338693] CHIP:DMG:                0x0 = 1, 
[1661909717.710873][338693:338693] CHIP:DMG:                0x1 = 1, 
[1661909717.710906][338693:338693] CHIP:DMG:                0x2 = 0, 
[1661909717.710939][338693:338693] CHIP:DMG:                0x3 = 1, 
[1661909717.710972][338693:338693] CHIP:DMG:                0x4 = true, 
[1661909717.711004][338693:338693] CHIP:DMG:            },
[1661909717.711033][338693:338693] CHIP:DMG:        },
[1661909717.711075][338693:338693] CHIP:DMG:        
[1661909717.711102][338693:338693] CHIP:DMG:    ],
[1661909717.711145][338693:338693] CHIP:DMG:    
[1661909717.711173][338693:338693] CHIP:DMG:    InteractionModelRevision = 1
[1661909717.711199][338693:338693] CHIP:DMG: },
[1661909717.711352][338693:338693] CHIP:DMG: AccessControl: checking f=1 a=c s=0x000000000001B669 t= c=0xFFF1_FC06 e=0 p=m
[1661909717.711412][338693:338693] CHIP:DMG: AccessControl: allowed
[1661909717.711449][338693:338693] CHIP:DMG: Received command for Endpoint=0 Cluster=0xFFF1_FC06 Command=0x0000_0000
[1661909717.711540][338693:338693] CHIP:ZCL: FaultInjection: Configure a fault of type: 1 and Id: 1 to be triggered deterministically
```
